src/cc_helpers.cc

```diff
@@ -761,6 +810,102 @@ bool certifier::framework::cc_trust_data::GetPlatformSaysAttestClaim(signed_clai
   return false;
 }
 
+bool certifier::framework::cc_trust_data::recertify_me(const string& host_name, int port, bool generate_new_key) {
+
+  if (generate_new_key) {
+
+    if (purpose_ == "authentication") {
+
+    // make app auth private and public key
+      if (public_key_algorithm_ == "rsa-2048") {
+        if (!make_certifier_rsa_key(2048,  &private_auth_key_)) {
+          printf("recertify_me: Can't generate App private key\n");
+          return false;
+        }
+      } else if (public_key_algorithm_ == "rsa-4096") {
+        if (!make_certifier_rsa_key(4096,  &private_auth_key_)) {
+          printf("recertify_me: Can't generate App private key\n");
+          return false;
+        }
+      } else if (public_key_algorithm_ == "ecc-384") {
+        if (!make_certifier_ecc_key(384,  &private_auth_key_)) {
+          printf("recertify_me: Can't generate App private key\n");
+          return false;
+        }
+      } else {
+          printf("recertify_me: Unsupported public key algorithm\n");
+          return false;
+      }
+
+      private_auth_key_.set_key_name("auth-key");
+      if (!private_key_to_public_key(private_auth_key_, &public_auth_key_)) {
+        printf("recertify_me: Can't make public Auth key\n");
```

Comment:
Same comment here and all places below to replace `recertify_me` with `__func__`.

It's a lot more maintainable that way.